tests/test_parser.py

```diff
@@ -21,44 +21,79 @@
 
 
 class TestsGetPage(unittest.TestCase):
-    def setUp(self):
-        pass
 
-    def tearDown(self):
-        pass
-
-    @patch('requests.get')
-    def test_perform_get_request_uses_provided_url_for_request(
-            self,
-            mock_request):
-        mock_request.return_value.status_code = 200
+    def test_perform_get_request_uses_provided_url_for_request(self):
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=200)
+            )
+        )
         exepted_url = 'https://smarsy.ua/'
-        perform_get_request(exepted_url)
-        mock_request.assert_called_with(exepted_url)
+        perform_get_request(session, exepted_url)
+        session.get.assert_called_with(url=exepted_url,
+                                       data=None, headers=None)
 
-    @patch('requests.get')
-    def test_perform_get_request_returns_expected_text_on_valid_request(
-            self,
-            mock_response):
+    def test_perform_get_request_returns_expected_text_on_valid_request(self):
         url = 'https://smarsy.ua/'
-        mock_response.return_value.status_code = 200
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=200)
+            )
+        )
         expected_text = 'This is login Page'
-        mock_response(url).text = expected_text
-        self.assertEqual(perform_get_request(url), expected_text)
+        session.get(url).text = expected_text
+        self.assertEqual(perform_get_request(session, url), expected_text)
 
-    @patch('requests.get')
-    def test_perform_get_requestresponse_with_status_code_404_raises_exception(
-            self,
-            mock_response):
+    def test_perform_get_request_resp_with_status_code_404_raises_exception(
+            self):
         url = 'https://smarsy.ua/'
-        mock_response.return_value.status_code = 404
-        self.assertRaises(requests.HTTPError, perform_get_request, url)
+        session = Mock(
+            get=MagicMock(
+                return_value=Mock(status_code=404)
+            )
+        )
+        self.assertRaises(requests.HTTPError, perform_get_request, session,
+                          url)
+
+    def test_perform_get_request_uses_provided_data_for_get_request(
+            self):
+        expected_data = 'data'
```

Comment:
is `data` needed for get requests?

Reply:
yes, to parse childs data page we need paste data in url like
https://smarsy.ua/parent.php', params=param, headers=headers

Reply:
if it is named differently in the original function (params) then it is better to use the same name

Reply:
variable data in perform_get_request changed to params